helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobs.java

```diff
@@ -499,35 +248,44 @@ static Builder builder(final Map<String, String> env) {
   }
 
   static Builder builder(final String profile, final Map<String, String> env) {
-    return builder(profile, env, HeliosClient.newBuilder());
+    return new Builder(profile, HeliosConfig.loadConfig("helios-testing"), env);
   }
 
-  static Builder builder(final String profile, final Map<String, String> env,
-                         final HeliosClient.Builder clientBuilder) {
-    return new Builder(profile, HeliosConfig.loadConfig("helios-testing"), env, clientBuilder);
+  @Override
+  public void close() {
+    final List<AssertionError> errors = new ArrayList<>();
+
+    for (final TemporaryJob job : jobs) {
+      // Undeploying a job doesn't guarantee the container will be stopped immediately.
+      // Luckily TaskRunner tells docker to wait 120 seconds after stopping to kill the container.
+      // So containers that don't immediately stop **should** only stay around for at most 120
+      // seconds.
+      errors.addAll(undeploy(job.job(), job.hosts()));
+    }
+
+    for (final AssertionError error : errors) {
+      log.error(error.getMessage());
+    }
```

Comment:
I'd like to deconstruct `TemporaryJobs` even more, right now it serves two purposes:

 - It glues everything together and is the place where you provide configuration, and
 - It keeps tracks of created jobs and undeploys them when `close()` is called

Why not separate those responsibilities out into separate classes (that are hopefully easier to reason about on their own)? E.g. a separate collection-of-temporary-jobs-that-takes-care-undeploying and a temporary-job-factory.